sdks/python/apache_beam/coders/coders.py

```diff
@@ -1383,22 +1375,74 @@ def from_runner_api_parameter(payload, components, context):
         write_state_threshold=int(payload))
 
 
-class RunnerAPICoderHolder(Coder):
+class ElementTypeHolder(typehints.TypeConstraint):
+  """A dummy element type for external coders that cannot be parsed in Python"""
+  def __init__(self, coder, context):
+    self.coder = coder
+    self.context = context
+
+
+class ExternalCoder(Coder):
   """A `Coder` that holds a runner API `Coder` proto.
 
   This is used for coders for which corresponding objects cannot be
   initialized in Python SDK. For example, coders for remote SDKs that may
   be available in Python SDK transform graph when expanding a cross-language
   transform.
   """
-  def __init__(self, proto):
-    self._proto = proto
 
-  def proto(self):
-    return self._proto
+  coder_count = 0
```

Comment:
I see this referenced half a dozen times below. Perhaps encapsulate the increment+return in a method `_unique_id` or similar?